proposals/xxxx-package-manager-performance-testing.md

```diff
@@ -0,0 +1,63 @@
+# Package Manager Performance Testing Conventions
+
+* Proposal: [SE-XXXX](xxx-package-manager-performance-testing.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This is a proposal for adding package manager support for building and running performance tests in release mode.
+
+## Motivation
+
+The package manager supports building and running tests using the `swift test` subtool. The sources are built in debug mode with swift compiler's
+`-enable-testing` flag allowing testable imports in test modules (which gives it internal level access to the imported module). 
+This works well for normal unit tests, however for performance tests we are usually interested in release build test results which takes advantages of compiler 
+optimizations.
+
+We can provide an option to compile and run the tests in release mode with `-enable-testing` flag. However that will disable some of the compiler optimizations, which 
+may be of interest to packages having very performance sensitive code (which critically depends on those optimization).
+
+Since performance tests usually take more time to run than other tests, there is also a requirement to have a way to run performance tests separate from unit tests.
+As a concrete example, package manager's own performance tests are disabled from running on Swift CI.
+
+## Proposed solution
+
+We propose to extend the test modules conventions and consider a directory as "performance test module" which has a suffix "PerformanceTests". The tests in these modules will only run when `--performance` flag is passed to `swift test`.
+
+This command will run performance tests in debug configuration:
+
+```
+$ swift test --performance
```

Comment:
I find the syntax ambiguous -- if I had never seen it before, I would wonder if it:

1. Enables performance tests?

2. Only runs performance tests?

3. Does something completely differently?

I like the way that if you read it as a sentence it does exactly what it says (especially if it only runs performance tests) but it does give me a little pause.